src/analyser.cpp

```diff
@@ -3308,16 +3306,10 @@ void Analyser::AnalyserImpl::analyseModel(const ModelPtr &model)
     }
 }
 
-AnalyserExternalVariablePtrs::const_iterator Analyser::AnalyserImpl::findExternalVariable(const ModelPtr &model,
-                                                                                          const std::string &componentName,
-                                                                                          const std::string &variableName) const
+AnalyserExternalVariablePtrs::const_iterator Analyser::AnalyserImpl::findExternalVariable(const VariablePtr &variable) const
 {
     return std::find_if(mExternalVariables.begin(), mExternalVariables.end(), [=](const auto &ev) {
-        auto variable = ev->variable();
-
-        return (owningModel(variable) == model)
-               && (owningComponent(variable)->name() == componentName)
-               && (variable->name() == variableName);
+        return ev->variable() == variable;
```

Comment:
It is now no longer a requirement that an external variable needs to be from the same model?

Reply:
Not so much that but rather the fact that before our API required a model, a component name, and a variable name to identify a variable and then check whether it was associated to one of our external variables.

Now, we directly pass a variable, which means that we can now directly check whether that variable is associated with one of our external variables.